### PR TITLE
[1.20.4] Feat: Fast vector

### LINKED ISSUE #45: [1.20.x] [All] [Feature]: Fast vector
### Feature

Implement a fast low-level vector with zero-allocation

We would benefit from
- Fast world traversal
- Less memory overhead from BlockPos and Vec3d/Vec3i
- DSL for user-friendly interaction

### Additional Information

X

### Pull Request

- [X] I will be submitting a pull request

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by emyfops**
Introduces a zero-allocation position tuple for low-level api world search

Closes #45

### CODE REVIEW

**Comment by emyfops** [comment]
Ignore the failed workflow, it is caused by the forge bootstrap that was fixed in e5ae0dd

File: common/src/main/kotlin/com/lambda/util/world/Position.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import net.minecraft.util.math.Vec3d
+import net.minecraft.util.math.Vec3i
+
+/**
+ * Represents a position in the world encoded as a long.
+ *
+ * [ X (26 bits) | Z (26 bits) | Y (12 bits) ]
+ *
+ * The position is encoded as a 64-bit long where the X and Z coordinates are stored in the most significant 26 bits
+ * and the Y coordinate is stored in the middle 12 bits. This encoding allows for a maximum world size of
```
Comment (Avanatiker) [inline common/src/main/kotlin/com/lambda/util/world/Position.kt]: middle -> least significant

File: common/src/main/kotlin/com/lambda/util/world/WorldUtils.kt [hunk 2]
```diff
@@ -196,22 +161,20 @@ object WorldUtils {
...
      */
     inline fun SafeContext.searchBlocks(
-        pos: Vec3i,
-        range: Vec3i,
-        pointer: MutableMap<BlockPos, Block>? = null,
-        predicate: (BlockState, BlockPos) -> Boolean = { _, _ -> true },
-        iterator: (BlockState, BlockPos, Int) -> Unit = { _, _, _ -> },
+        pos: FastVector,
+        range: FastVector,
+        step: FastVector = 274945015809L,
```
Comment (Avanatiker) [inline common/src/main/kotlin/com/lambda/util/world/WorldUtils.kt]: magic number?

File: common/src/main/kotlin/com/lambda/util/world/WorldUtils.kt [hunk 3]
```diff
@@ -227,22 +190,20 @@ object WorldUtils {
...
      */
     inline fun <reified T : Fluid> SafeContext.searchFluids(
-        pos: Vec3i,
-        range: Vec3i,
-        pointer: MutableMap<BlockPos, T>? = null,
-        iterator: (FluidState, BlockPos, Int) -> Unit = { _, _, _ -> },
-        predicate: (FluidState, BlockPos) -> Boolean = { _, _ -> true },
+        pos: FastVector,
+        range: FastVector,
+        step: FastVector = 274945015809L,
```
Comment (Avanatiker) [inline common/src/main/kotlin/com/lambda/util/world/WorldUtils.kt]: magic number? use constants instead and explain them

File: common/src/main/kotlin/com/lambda/util/world/WorldUtils.kt [hunk 4]
```diff
@@ -252,16 +213,27 @@ object WorldUtils {
...
+     * @param step The step to increment the position by.
      * @param iterator Iterator to perform operations on each position.
      */
     inline fun iteratePositions(
-        pos: Vec3i,
-        range: Vec3i,
-        iterator: (BlockPos, Int) -> Unit,
+        pos: FastVector,
+        range: FastVector,
+        step: FastVector = 274945015809L,
```
Comment (Avanatiker) [inline common/src/main/kotlin/com/lambda/util/world/WorldUtils.kt]: magic number?

**Review by Avanatiker — CHANGES_REQUESTED** [review]

**Pull request merged**